File: moviecutter.py
import json
import requests
import csv
import re
from unidecode import unidecode
from datetime import datetime
from os.path import expanduser

# set the output file's destination
# AMEND THIS TO CHANGE WHERE OUTPUT WILL GO
writepath = expanduser('~/Downloads/moviecutter.csv')

# streaming services to check
# key names are same as CanIStreamIt? for easier iteration
# services = ['netflix_instant', 'epix', 'crackle', 'redboxinstant_subscription',
#			'amazon_prime_instant_video', 'hulu_movies', 'hitbliss_streaming',
#			'youtube_free', 'snagfilms', 'streampix', 'hbo', 'showtime', 'cinemax',
#			'starz', 'encore', 'xfinity_free']

# simple function for cleaning up titles by removing parens
def cleantitle(title):
	words = title.split()
	
	ret = ""
	for word in words:
		if (word[0]!='(' and word[-1]!=')'):
		 ret += unidecode(word) + ' '
	
	# get rid of last space
	ret = ret[0:-1]
		 
	return ret

# get the full Metacritic list
r = requests.get('http://www.kimonolabs.com/api/7oxdwnlm?apikey=8df3757b504a8513ce1380c13b356cd9')
metacritic = r.json()["results"]["collection1"]

print "Metascores received\n" + str(datetime.now().time())	
print "Checking tomatometer scores and streaming services; this could take a couple of minutes..."

headers = ["name", "metascore", "tomatometer"]
csvlist = [headers]


for (index, film) in enumerate(metacritic):
	score = int(film["metascore"])
	
	# limit to 80+ scorers
	if score>=80:
	
		
		title = cleantitle(film["title"]["text"])
		row = [title]
		row.append(int(film["metascore"]))
		
		# get tomatometer and canistreamit data
		try:
			tomatometer = requests.get('http://api.rottentomatoes.com/api/public/v1.0/movies.json?apikey=t33zpxft4pdupfnqnq4mbjsb&q='
										+ title + '&page_limit=1').json()["movies"][0]["ratings"]["critics_score"]
			
			row.append(tomatometer)
			
		##	k = requests.get('http://www.canistream.it/services/search', params={'movieName': title}).content
			
		##	movie = canistreamit.search(title)[0]['_id']
		##	print movie
		##	for service in services:
		##		try:
		##			if service in canistreamit.streaming(movie):
		##				row.append("True")
		##			elif service in canistreamit.xfinity(movie):
		##				row.append("True")
		##			else:
		##				row.append("False")
		##		except requests.exceptions.ConnectionError:
		##			continue

		except:
			continue
				
		csvlist.append(row)

print "Tomatometers received\n" + str(datetime.now().time())

	
# write it out
with open(writepath, 'wb') as csvfile:
	writer = csv.writer(csvfile, delimiter=',', quoting=csv.QUOTE_NONNUMERIC)
	writer.writerows(csvlist)

print "CSV file written"

File: moviecutter.go
package main

import (
	"fmt",
	"io",
	"csv",
	"json",
	"filepath",
	"net/http",
	"strings",
	"strconv",
	"regexp",
	"canistreamit"
)

// change value if you want to export to a different path
writepath, err := filepath.Abs("~/Downloads/moviecutter_go.csv")

// services names
services := ["netflix_instant", "epix", "crackle", "redboxinstant_subscription",
			"amazon_prime_instant_video", "hulu_movies", "hitbliss_streaming",
			"youtube_free", "snagfilms", "streampix", "hbo", "showtime", "cinemax",
			"starz", "encore", "xfinity_free"]

// Kimono API key
kimono_api_key := "8df3757b504a8513ce1380c13b356cd9"
kimono_params := make(map[string][string])
kimono_params["apikey"] := kimono_api_key
metacritic_url := "http://www.kimonolabs.com/api/7oxdwnlm"

// Rotten Tomatoes API key
rt_api_key := "t33zpxft4pdupfnqnq4mbjsb"

// title cleaning 
func CleanTitle(title string) string {
	
	// regex matching parenthesized words or multi-word phrases
	// luckily (500) Days of Summer is only a 76 on Metacritic
	re := regexp.MustCompile("([^)]*)")
	
	return re.ReplaceAllString(title, "")
	
}

// get the full Metacritic list
metacritic_response = JSONGet(metacritic_url,kimono_params)["results"]["collection1"]

for i, item := range metacritic_revenge {
	fmt.Printf(item)
}

// reorient it as a key-value pair
meta := make(map[string][string])

// use goroutines to reformat the response
for index, film := range metacritic_response {
	go func (film map[string][string]) {
		meta[CleanTitle(film["title"]["text"])] = strconv.ParseInt(film["metascore"], 10, 0)
	} (film)
}

// use goroutines to exclude sub-80 scorers
for i in meta.keys():
	if meta[i] >= 80:
		meta2[i] = meta[i]
meta = meta2


 